File: plugins/OrchidPosts/src/main/java/com/eden/orchid/posts/permalink/PermalinkPathType.java
package com.eden.orchid.posts.permalink;

import com.eden.orchid.api.registration.Prioritized;
import com.eden.orchid.posts.pages.PostPage;

public abstract class PermalinkPathType extends Prioritized {

    public abstract boolean acceptsKey(PostPage post, String key);

    public abstract String format(PostPage post, String key);

}

File: plugins/OrchidPosts/src/main/kotlin/com/eden/orchid/posts/permalink/PermalinkPathType.kt
package com.eden.orchid.posts.permalink

import com.eden.orchid.api.registration.Prioritized
import com.eden.orchid.api.theme.pages.OrchidPage

abstract class PermalinkPathType(priority: Int) : Prioritized(priority) {

    abstract fun acceptsKey(page: OrchidPage, key: String): Boolean

    abstract fun format(page: OrchidPage, key: String): String?

}
